packages/beacon-node/src/chain/blocks/verifyBlock.ts

```diff
@@ -66,16 +66,30 @@ export async function verifyBlocksInEpoch(
   const abortController = new AbortController();
 
   try {
-    const [{postStates, proposerBalanceDeltas}, , segmentExecStatus] = await Promise.all([
+    const [segmentExecStatus, {postStates, proposerBalanceDeltas}] = await Promise.all([
+      // Execution payloads
+      verifyBlocksExecutionPayload(
+        {...this, metrics: this.metrics},
```

Comment:
Not sure if spreading `this` is a good idea, that's the entire BeaconChain object. Can you find a different pattern?

Reply:
I'd make `metrics` as public in `IBeaconChain`, similar to `logger`